pkg/config/spec.go

```diff
@@ -1025,3 +1015,28 @@ func DetectPreConfiguredDevice(logger sdkTypes.KairosLogger) (string, error) {
 
 	return "", nil
 }
+
+// findUpgradeEntryFromConfig checks the passed config for one of the following ways
+// to set the upgrade entry.
+// - One way is to use the cli arg "--recovery" which makes config.upgrade.entry be "recovery"
+// - Another way is by setting the cli arg "boot-entry" which set it to the specified value.
+// - Lastly, user can set "upgrade.recovery: true" in the kairos config, which should result in entry being "recovery".
+func findUpgradeEntryFromConfig(cfg *Config) string {
+	if _, ok := cfg.Config["upgrade"]; !ok {
+		return ""
+	}
+
+	// check value from --recovery and --boot-entry
+	if entryIface, ok := cfg.Config["upgrade"].(collector.Config)["entry"]; ok {
```

Comment:
What I meant was, we don't need to handle this manually here. The `umarshallFullSpec` will read the `upgrade.entry` and set it in the UpgradeSpec correctly. Nothing needs this value between this code here and the call to `umarshallFullSpec` so it should be ok to skip I think. Also the test still passes which is a proof that it still works (after I removed this part here).

Reply:
ahh, could be could be. we did that becuase we first called the calculate size and that filled the size of the given entry, recovery or active. Indeed it may now not be needed!

Easy to check as well, tests should check that we trigger the recovery upgrade or the active one, so good to me!